### PR TITLE
[CWS] replace usages of LockOSThread with syscall tester in functional tests

### PR DESCRIPTION
**Pull request by paulcacheux**
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/tests/syscall_tester/c/syscall_tester.c [hunk 1]
```diff
@@ -214,6 +214,31 @@ int test_splice() {
...
+    }
+
+    if (setreuid(1, 1) != 0) {
+        fprintf(stderr, "setreuid failed");
+        return EXIT_FAILURE;
+    }
+
+    fprintf(stderr, "mkdir path: %s\n", argv[1]);
+    if (mkdirat(0, argv[1], 0777) == 0) {
+        fprintf(stderr, "mkdirat succeeded event though we expected it to fail");
```
Comment (Gui774ume) [inline pkg/security/tests/syscall_tester/c/syscall_tester.c:235]: even 😄

File: pkg/security/tests/syscall_tester/c/syscall_tester.c [hunk 2]
```diff
@@ -214,6 +214,31 @@ int test_splice() {
...
+        fprintf(stderr, "setregid failed");
+        return EXIT_FAILURE;
+    }
+
+    if (setreuid(1, 1) != 0) {
+        fprintf(stderr, "setreuid failed");
+        return EXIT_FAILURE;
+    }
+
+    fprintf(stderr, "mkdir path: %s\n", argv[1]);
```
Comment (Gui774ume) [inline pkg/security/tests/syscall_tester/c/syscall_tester.c:233]: debug log ? 😄

Comment (paulcacheux) [reply pkg/security/tests/syscall_tester/c/syscall_tester.c:233]: ah yep ahah

**Review by Gui774ume — APPROVED** [review]
🚢

**Pull request merged**